collections/iter.go

```diff
@@ -130,39 +130,48 @@ func (r *Range[K]) RangeValues() (start, end *RangeKey[K], order Order, err erro
 	return r.start, r.end, r.order, nil
 }
 
-// iteratorFromRanger generates an Iterator instance, with the proper prefixing and ranging.
-// a nil Ranger can be seen as an ascending iteration over all the possible keys.
-func iteratorFromRanger[K, V any](ctx context.Context, m Map[K, V], r Ranger[K]) (iter Iterator[K, V], err error) {
+// parseRangeInstruction converts a Ranger into start bytes, end bytes and order of a store iteration.
+func parseRangeInstruction[K any](prefix []byte, keyCodec codec.KeyCodec[K], r Ranger[K]) ([]byte, []byte, Order, error) {
```

Comment:
this logic was part of `iteratorFromRanger` but I separated it because only this part was needed for the `Clear` method.